### PR TITLE
Allows resetting of uplink items in case one gets deleted, adds null checking to get_uplink_items()

### PR DESCRIPTION
**Pull request by silicons**
You know, when admins have to delete one in emergencies, and then realize that everything's broken because the list now has a null..

### CODE REVIEW

**Comment by coiax** [comment]
When would they need to delete one?

**Comment by kevinz000** [comment]
@coiax when there's a security risk to the server for one.

File: code/modules/uplink/uplink_item.dm [hunk 1]
```diff
@@ -117,6 +123,11 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 			to_chat(H, "\The [A] materializes onto the floor.")
 	return 1
 
+/datum/uplink_item/Destroy()
+	if(src in uplink_items)
+		uplink_items -= src	//Take us out instead of leaving a null!
+	..()
```
Comment (lzimann) [inline code/modules/uplink/uplink_item.dm]: return ..()

File: code/modules/uplink/uplink_item.dm [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
-			uplink_items[I.category][I.name] = I
+		initialize_global_uplink_items()
 
 	var/list/filtered_uplink_items = list()
 	var/list/sale_items = list()
 
 	for(var/category in uplink_items)
 		for(var/item in uplink_items[category])
 			var/datum/uplink_item/I = uplink_items[category][item]
+			if(!I || !istype(I))
```
Comment (lzimann) [inline code/modules/uplink/uplink_item.dm]: istype is not really necessary here, if something that is not an uplink_item is on this list, then there's something really wrong with it

Comment (Cyberboss) [inline code/modules/uplink/uplink_item.dm]: Also, just `!istype(I)` would have sufficed

**Comment by kevinz000** [comment]
is the istypes really that bad to leave in the proc isn't used much and when it's used it's either generating a new list or when something's bugged up. when globals become editable or whatever it's probably good to have such a check.

File: code/modules/uplink/uplink_item.dm [hunk 3]
```diff
@@ -20,7 +20,7 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	for(var/category in uplink_items)
 		for(var/item in uplink_items[category])
 			var/datum/uplink_item/I = uplink_items[category][item]
-			if(!I || !istype(I))
+			ifistype(I))
```
Comment (coiax) [inline code/modules/uplink/uplink_item.dm]: `if(istype(I))`

File: code/modules/uplink/uplink_item.dm [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
-			uplink_items[I.category][I.name] = I
+		initialize_global_uplink_items()
 
 	var/list/filtered_uplink_items = list()
 	var/list/sale_items = list()
 
 	for(var/category in uplink_items)
 		for(var/item in uplink_items[category])
 			var/datum/uplink_item/I = uplink_items[category][item]
+			if(istype(I))
```
Comment (AnturK) [inline code/modules/uplink/uplink_item.dm]: !

**Review by AnturK — DISMISSED** [review]

**Pull request merged**